src/components/pad-view-only/pad-view-only.tsx

```diff
@@ -4,94 +4,56 @@
+import { useMarkdownContent } from '../../hooks/common/use-markdown-content'
 import { ApplicationState } from '../../redux'
-import { setDocumentContent, setDocumentMetadata } from '../../redux/document-content/methods'
-import { extractNoteTitle } from '../common/document-title/note-title-extractor'
+import { setNoteMetadata, updateNoteTitleByFirstHeading } from '../../redux/note-content/methods'
 import { MotdBanner } from '../common/motd-banner/motd-banner'
 import { ShowIf } from '../common/show-if/show-if'
 import { AppBar, AppBarMode } from '../editor/app-bar/app-bar'
 import { DocumentIframe } from '../editor/document-renderer-pane/document-iframe'
 import { EditorPathParams } from '../editor/editor'
-import { YAMLMetaData } from '../editor/yaml-metadata/yaml-metadata'
+import { useLoadNoteFromServer } from '../editor/useLoadNoteFromServer'
 import { DocumentInfobar } from './document-infobar'
+import { ErrorWhileLoadingNoteAlert } from './ErrorWhileLoadingNoteAlert'
+import { LoadingNoteAlert } from './LoadingNoteAlert'
 
 export const PadViewOnly: React.FC = () => {
-  const { t } = useTranslation()
-  const { id } = useParams<EditorPathParams>()
-  const untitledNote = t('editor.untitledNote')
-  const [documentTitle, setDocumentTitle] = useState(untitledNote)
-  const [noteData, setNoteData] = useState<Note>()
-  const [error, setError] = useState(false)
-  const [loading, setLoading] = useState(true)
-  const noteMetadata = useRef<YAMLMetaData>()
-  const firstHeading = useRef<string>()
-
-  const updateDocumentTitle = useCallback(() => {
-    const noteTitle = extractNoteTitle(untitledNote, noteMetadata.current, firstHeading.current)
-    setDocumentTitle(noteTitle)
-  }, [untitledNote])
-
-  const onFirstHeadingChange = useCallback((newFirstHeading: string | undefined) => {
-    firstHeading.current = newFirstHeading
-    updateDocumentTitle()
-  }, [updateDocumentTitle])
 
-  const onMetadataChange = useCallback((metaData: YAMLMetaData | undefined) => {
-    noteMetadata.current = metaData
-    setDocumentMetadata(metaData)
-    updateDocumentTitle()
-  }, [updateDocumentTitle])
-
-  useEffect(() => {
-    getNote(id)
-      .then(note => {
-        setNoteData(note)
-        setDocumentContent(note.content)
-      })
-      .catch(() => setError(true))
-      .finally(() => setLoading(false))
-  }, [id])
+  useTranslation()
+  const { id } = useParams<EditorPathParams>()
 
   useApplyDarkMode()
-  useDocumentTitle(documentTitle)
-  const markdownContent = useSelector((state: ApplicationState) => state.documentContent.content)
+  useDocumentTitleWithNoteTitle()
+
+  const onFirstHeadingChange = useCallback(updateNoteTitleByFirstHeading, [])
+  const onMetadataChange = useCallback(setNoteMetadata, [])
+  const [error, loading] = useLoadNoteFromServer()
+  const markdownContent = useMarkdownContent()
+  const noteMetaData = useSelector((state: ApplicationState) => state.noteContent)
```

Comment:
`noteContent` is actually a better name as it contains the content besides the metadata as well.

Reply:
What about "useNoteMarkdownContent" ? I want to include the "markdown" to make the hook more specific.